Move Build Targets page to CLI guides (#163)
ember-learn/cli-guides: guides/advanced-use/build-targets.md

Ember CLI by default uses [Babel.js](https://babeljs.io/) to allow you to use tomorrow's JavaScript, today.

It will ensure that you can use the newest features in the language and know that they will be transformed to JavaScript that can run in every browser you support.

> That usually means generating ES5-compatible code that can work on any modern browser, back to Internet Explorer 11.

Reviewer: ```suggestion
That usually means generating ES5-compatible code that can work on any legacy browser, back to Internet Explorer 11.
```

Author: I think "modern" here is more correct, but I can see how it's confusing. How about just removing this sentence? The previous sentence ends "every browser you support", and I think that should be enough. I don't see a need to call out IE11 specifically

It can figure out automatically what features are supported by the browsers you are targeting,
and apply the minimum set of transformations possible to your code.

If you open `config/targets.js`, you will find the following code:

```javascript {data-filename=config/targets.js}
const browsers = [
  'last 1 Chrome versions',
  'last 1 Firefox versions',
  'last 1 Safari versions'
];

const isCI = !!process.env.CI;
const isProduction = process.env.EMBER_ENV === 'production';

if (isCI || isProduction) {
  browsers.push('ie 11');
}

module.exports = {
  browsers
};
```

That default configuration has two parts.
In `production` environment it matches the wider set of browsers that Ember.js itself supports.
In non-production builds a narrower subset of browsers is used to provide the best experience
for developers to debug code that is much closer to what they actually wrote.

However, if your app does not need to support IE anymore, you could change it to:

```javascript {data-filename=config/targets.js}
module.exports = {
  browsers: [
    'last 1 edge versions',
    'last 1 Chrome versions',
    'last 1 Firefox versions',
    'last 1 Safari versions'
  ]
};
```

> You are left with browsers that have full support of ES2015 and ES2016.

Reviewer: ```suggestion
You are left with browsers that have the full support of the latest JavaScript features.
```
Because we are talking about async/await in the below lines which is from ES2017

If you inspect the compiled code, you will see that some features are not compiled to ES5 code anymore, such as arrow functions and async/await.

> This feature is backed by [Browserlist](https://github.com/ai/browserslist) and [Can I Use](http://caniuse.com/).

Reviewer: ```suggestion
This feature is backed by [Browserslist](https://github.com/ai/browserslist) and [Can I Use](https://caniuse.com/).
```
